src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py

```diff
@@ -602,7 +614,7 @@ def _predict_with_probabilities(self, input_dataset, batch_size=64,
                 if verbose and (dataset_size >= 5
                     and cur_time > last_time + 30 or done):
                     print('Predicting {cur_n:{width}d}/{max_n:{width}d}'.format(
-                        cur_n = index + 1 if not done else index,
+                        cur_n = index + 1 if index + 1 > dataset_size - 1 else index,
```

Comment:
I'm having a hard time understanding this line change.

Is `done` not actually getting set to `True` when things are done?

Reply:
So I had made this change in my earlier PR with not done in my earlier PR but that hadn't fixed this issue. If you see index is being incremented by split_length and then checks if it is equal to dataset_size - 1. So basically index skips over the dataset_size - 1 value for done to be true. But I am not fixing it correctly I should have just make checked if index > dateset_size -1 and kept this code as it is. Does this make sense?

Reply:
That sounds good. Let's not change this line and let's just make sure that `done` is getting set correctly.

Reply:
Yes, got it. I will change it.

Reply:
This code reads weirdly to me. AFAICT, `index` exists only to track how many instances have been processed. I would be in favor of just renaming it `cur_n` directly. I also think we can just remove the variable `done` entirely. It looks to me like the current computation of `done` is incorrect, it should be:
```
if cur_n >= dataset_size:
  cur_n = dataset_size
```
And then instead of testing `done` you can just test `cur_n >= dataset_size`.